### ISSUE TITLE
[telemetry] Failed-login throttling metrics for LoginThrottlePerIP

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#44]
## Goal

Expose failed-login throttling per IP as ph-telemetry metrics, so that brute-force and credential-stuffing activity is visible outside the log file.

## Where

`ph-oton-security/src/main/java/com/helger/photon/security/login/LoginThrottlePerIP.java`

```java
public int onFailedLogin (@NonNull @Nonempty final String sIP)
public EChange onSuccessfulLogin (@Nullable final String sIP)
public int getFailedLoginCount (@Nullable final String sIP)
public void clear ()
```

The class keeps a `ManualCache <String, Integer>` (`CACHE_NAME = "login-failed-per-ip"`) with a `DEFAULT_TIME_TO_LIVE` of one hour. Today none of this is observable.

The consumer is `ph-oton-core/src/main/java/com/helger/photon/core/login/AbstractLoginManager.java` (`checkUserAndShowLogin (...)`), which applies `getFailedLoginWaitingTime ()` as an artificial delay after a failed login. Note that `LoginThrottlePerIP` was moved from `com.helger.photon.core.login` to `com.helger.photon.security.login` in 10.5.0, so it now lives in `ph-oton-security`.

## Cardinality and privacy — hard rules

**The IP address must never become a metric attribute.** It is unbounded and it is personal data under GDPR. This is the single most important constraint in this issue.

What is safe:
- a plain count of throttle events (no attributes at all),
- a gauge over the *size* of the throttle cache — i.e. how many distinct IPs currently have failed logins on record.

If a per-IP breakdown is ever wanted, that belongs in the audit log or in a span attribute on an individual login attempt, never in a metric.

## What to emit

Extend `com.helger.photon.security.CSecurityTelemetry` (created by the `LoggedInUserManager` login-metrics issue; create it here if that one has not landed yet):

| Constant | Value |
|---|---|
| `METRIC_LOGIN_THROTTLE_FAILED` | `photon.security.throttle.failed` |
| `METRIC_LOGIN_THROTTLE_DELAY` | `photon.security.throttle.delay` |
| `METRIC_LOGIN_THROTTLE_TRACKED_IPS` | `photon.security.throttle.tracked` |
| `UNIT_LOGIN` | `{login}` |
| `UNIT_IP` | `{ip}` |
| `UNIT_MILLIS` | `ms` |

- **Counter** `photon.security.throttle.failed` — incremented in `onFailedLogin (...)`, no attributes.
- **Histogram** `photon.security.throttle.delay` in ms — the artificial wait actually applied, recorded in `AbstractLoginManager.checkUserAndShowLogin (...)` where `aRealWaitDuration` is computed. This is the metric that shows an attack in progress: the aggregate delay climbs.
- **Gauge** `photon.security.throttle.tracked` — number of distinct IPs currently in the cache. Check what `ManualCache` exposes for its current size before committing to this one; if there is no cheap size accessor, drop the gauge rather than adding one to `ManualCache`.

## Gauge lifecycle caveat

`TelemetryMetrics.gauge (...)` returns a `Closeable` `ITelemetryGauge`. `LoginThrottlePerIP` is an ordinary instance (held by `AbstractLoginManager`), not a singleton, so the gauge must be tied to the instance lifecycle, not to a `static final` field. If that turns out awkward, prefer skipping the gauge — the counter and the histogram carry most of the value.

## POM

`ph-oton-security/pom.xml` needs a new dependency if the login-metrics issue has not already added it (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

`ph-oton-core` needs it too for the delay histogram; it is available transitively via `ph-oton-mgrs` but should be declared directly.

## Acceptance criteria

- [ ] No IP address on any metric attribute.
- [ ] Counter increments once per `onFailedLogin (...)`.
- [ ] Delay histogram records the delay actually applied in `AbstractLoginManager`.
- [ ] Unit test with recording SPIs installed via `TelemetryMetrics.install (...)` and reset to `null` in `@After`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-security/.../security/login/LoginThrottleMetrics.java` | new, public - counter + histogram |
| `ph-oton-security/.../security/login/LoginThrottlePerIP.java` | modified - one counter increment |
| `ph-oton-core/.../core/login/AbstractLoginManager.java` | modified - one histogram record |
| `ph-oton-security/src/test/.../LoginThrottleTelemetryTest.java` | new, 2 tests |

The metric names live in `CSecurityTelemetry`, which #43 introduced. `LoginThrottleMetrics` is public so that `AbstractLoginManager` in ph-oton-core can record the delay into the very same instrument - no emitter class, because the whole emission is two lines across two modules.

### Emission

* Counter `photon.security.throttle.failed` in `onFailedLogin (...)`, **no attributes at all**.
* Histogram `photon.security.throttle.delay` in `AbstractLoginManager.checkUserAndShowLogin (...)`, recorded from `aRealWaitDuration.toMillis ()` right where the delay is computed and immediately before `ThreadHelper.sleep (...)` - so it is the wait that is really applied, not the configured base value.

### Privacy

The IP address never touches a metric attribute. `LoginThrottleTelemetryTest.testFailedLoginIsCounted` calls `onFailedLogin` for two different IPs and asserts that **every** recorded measurement has an empty attribute map - the test fails if an attribute is ever added. The class Javadoc of `LoginThrottleMetrics` states the reason.

### The tracked-IPs gauge was skipped

The issue permitted this ("prefer skipping the gauge"), and it is the right call here:

* `ManualCache` *does* have a cheap `size ()` (inherited from `AbstractMapBasedCache`), so the data would be available;
* but `LoginThrottlePerIP` is a plain instance held by `AbstractLoginManager`, with no destruction hook, while `ITelemetryGauge` is `Closeable`. A gauge created in the constructor could never be closed and would leak - exactly the lifecycle trap #43 had to solve with `onAfterInstantiation` / `onDestroy` on the global singleton.

If a per-deployment "how many distinct IPs are currently throttled" signal is wanted later, the clean way is to give `LoginThrottlePerIP` an explicit lifecycle and create the gauge there. A cheaper alternative that has no lifecycle problem at all would be a histogram over the *consecutive* failure count returned by `onFailedLogin (...)` - it shows attack depth per attempt without any per-IP dimension. Neither was added here.

### Verification

Whole ph-oton-security suite: 49/49 green, including the pre-existing `LoginThrottlePerIPTest`. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.5.1 - work in progress`.

Note on the test setup: the recording meter is registered via the SPI mechanism rather than `TelemetryMetrics.install (...)`, because `LoginThrottlePerIPTest` runs earlier in the same JVM and would otherwise bind `LoginThrottleMetrics` to the no-op meter first. See #43 for the details.

**Comment by phax** [comment]
### Follow-up: the tracked-IPs gauge is now implemented

The gauge was skipped here because `LoginThrottlePerIP` was a plain instance owned by `AbstractLoginManager`, with no destruction hook to close a `Closeable` `ITelemetryGauge`.

#56 removed that obstacle - `LoginThrottlePerIP` is now a global singleton with `onAfterInstantiation` / `onDestroy` - so the gauge from this issue was added there:

* `photon.security.throttle.tracked` (unit `{ip}`), an observable gauge over the new `LoginThrottlePerIP.getTrackedIPCount ()`, which reads `ManualCache.size ()` under the read lock.
* Created in `onAfterInstantiation`, closed in `onDestroy`. No attributes - only the count of tracked IPs leaves the process, never an address.
* New constants `CSecurityTelemetry.METRIC_LOGIN_THROTTLE_TRACKED_IPS` and `UNIT_IP`, and the factory `LoginThrottleMetrics.createTrackedIPsGauge (LongSupplier)`.
* Covered by `LoginThrottleTelemetryTest.testTrackedIPsGauge`.

All three instruments this issue specified are therefore present in **10.6.0**. Nothing left open here.
